setlexsem/generate/generate_prompts.py

```diff
@@ -0,0 +1,248 @@
+# coding: utf-8
+
+import argparse
+import ast
+import logging
+import os
+import random
+from collections.abc import Iterable
+from itertools import product
+
+import pandas as pd
+from tqdm import tqdm
+
+from setlexsem.constants import PATH_PROMPTS_ROOT, PATH_ROOT
+from setlexsem.generate.prompt import (
+    PromptConfig,
+    get_ground_truth,
+    get_prompt,
+)
+from setlexsem.generate.sample import (
+    BasicNumberSampler,
+    BasicWordSampler,
+    DeceptiveWordSampler,
+    DecileWordSampler,
+    OverlapSampler,
+    Sampler,
+)
+from setlexsem.generate.utils_data_generation import load_generated_data
+from setlexsem.utils import get_prompt_file_path, read_config
+
+
+def replace_none(list_in):
+    return [None if x == "None" else x for x in list_in]
+
+
+# define argparser
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config_path",
+        type=str,
+        help="save files to disk",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def create_prompt(
+    sampler: Sampler,
+    prompt_config: PromptConfig,
+    num_runs=100,
+    add_roles=False,  # Claude Instant
+):
+    results = 0
+    prompt_and_ground_truth = []
+    for i in tqdm(range(num_runs)):
+        # create two sets from the sampler
+        if isinstance(sampler, Iterable):
+            # get next set from generator
+            A, B = next(sampler)
+            A = ast.literal_eval(A)
+            B = ast.literal_eval(B)
+        else:
+            # generate next set
+            A, B = sampler()
+
+        # Assign operation to the prompt_config
+        prompt = get_prompt(
+            A,
+            B,
+            prompt_config,
+            add_roles=add_roles,
+        )
+
+        ground_truth = get_ground_truth(prompt_config.operation, A, B)
+
+        prompt_and_ground_truth.append(
+            {
+                "prompt": prompt,
+                "ground_truth": ground_truth,
+                **prompt_config.to_dict(),
+            }
+        )
+
+    return prompt_and_ground_truth
+
+
+def main(config_file):
+    LOGGER = logging.getLogger(__name__)
```

Comment:
Do these variable names have to be all caps? That's usually reserved for global variables.

Reply:
no, we need to do some cleaning and fix some of these mismatches. I'll keep it as a very near future separate commit.